### PR TITLE
feat: Add changelog trimming functionality for readme.txt files

### PR DESCRIPTION
**Pull request by Camwyn**
# 🚀 feat: Add intelligent changelog trimming for WordPress plugin `readme.txt` files

## 📋 Overview

This PR introduces comprehensive changelog trimming functionality to ensure all TEC plugin `readme.txt` files comply with WordPress plugin directory requirements while preserving the most recent and relevant changelog information.

## 🎯 Problem Statement

WordPress plugin directory has strict guidelines for `readme.txt` file sizes, and our changelog sections often exceed recommended word limits (typically 5,000 words). Previously, we had no automated way to:
- Monitor changelog section word counts
- Intelligently trim older entries while preserving recent ones
- Maintain proper formatting and links to full changelogs
- Handle this consistently across all TEC plugins

## 💡 Solution

This implementation provides:

✅ **Intelligent Section-Based Trimming**: Removes complete release entries (oldest first) rather than cutting partial content
✅ **Configurable Word Limits**: Default 5,000 words with easy configuration per plugin
✅ **Smart Preservation**: Always keeps newest entries and never creates broken/partial releases
✅ **Automatic Link Addition**: Adds "See changelog for all versions" links when trimming occurs
✅ **Plugin-Specific Configuration**: Per-plugin changelog URLs via `package.json`
✅ **Cross-Platform Compatibility**: Works on both Linux (GitHub Actions) and macOS (local development)

### Core Functionality
- **`templates/bin/trim-readme-changelog.sh`** *(NEW)*: 376-line Bash script that handles intelligent changelog trimming
- **`templates/bin/process-changelog.sh`** *(ENHANCED)*: Integration with trimming functionality and package.json configuration support
- **`.github/actions/process-changelog/action.yml`** *(ENHANCED)*: New `changelog-full-url` input parameter

### Documentation & Configuration
- **`.github/actions/process-changelog/README.md`** *(NEW)*: Comprehensive 308-line documentation with usage examples, troubleshooting, and Mermaid workflow diagrams
- **`examples/package-json-changelog-config.md`** *(NEW)*: Configuration examples for all TEC plugins
- **`.github/sync.yml`** *(UPDATED)*: Include new trimming script in repository synchronization
- **`templates/workflows/release-process-changelog.yml`** *(UPDATED)*: Support for configurable changelog URLs

### Smart Trimming Logic
```
Original: 50 entries (8,500 words) → Trimmed: 7 entries (966 words) + link
```

- **Boundary Detection**: Uses precise release header parsing (`= [VERSION] DATE =`)
- **Word Count Exclusion**: Headers don't count toward word limits
- **Complete Entry Preservation**: Never creates partial or broken changelog entries
- **Oldest-First Removal**: Preserves most recent and relevant information

### Plugin-Specific Configuration
```json
{
  "tec": {
    "changelog_url": "https://evnt.is/1b5k"
  }
}
```

**Configuration Priority:**
1. `package.json` (plugin-specific, recommended)
2. Action parameter (workflow override)
3. Default fallback (`https://evnt.is/1b5k`)

### Workflow Integration
- Automatically runs during changelog processing
- Only modifies `readme.txt` (never touches `changelog.md`)
- Provides detailed statistics and feedback
- Handles errors gracefully with warnings

## 🧪 Testing

✅ **Local Testing Completed**: Verified with real TEC changelog data
✅ **Cross-Platform Tested**: Works on both Linux and macOS
✅ **Edge Cases Handled**: Empty sections, missing files, malformed headers
✅ **Configuration Tested**: package.json overrides and fallback behavior

### Test Results
```
Original entries: 50
Kept entries: 7
Total word count: 966
Was trimmed: true
```

## 📦 Deployment Strategy

1. **Phase 1**: Merge this PR to enable functionality in actions repository
2. **Phase 2**: Sync deployment will automatically push scripts to all TEC plugin repositories:
   - `the-events-calendar/tribe-common`
   - `the-events-calendar/the-events-calendar`
   - `the-events-calendar/events-pro`
   - `the-events-calendar/events-eventbrite`
   - `the-events-calendar/events-community`
   - `the-events-calendar/events-filterbar`
   - `the-events-calendar/event-tickets`
   - `the-events-calendar/event-tickets-plus`

3. **Phase 3**: Plugin teams can configure their specific changelog URLs in `package.json`

## 🔍 Affected Workflows

- **Release: Process Changelogs** - Enhanced with trimming functionality
- **All release workflows** - Will automatically benefit from trimmed readme.txt files
- **Plugin repository sync** - Will deploy trimming scripts ecosystem-wide

## 📚 Documentation

- Complete README with usage examples and troubleshooting
- Mermaid workflow diagrams showing processing flow
- Configuration examples for all TEC plugins
- Error handling and debugging guidance

## 🚦 Breaking Changes

❌ **None** - This is a purely additive enhancement that:
- Doesn't modify existing changelog generation logic
- Only affects `readme.txt` formatting (never `changelog.md`)
- Maintains backward compatibility
- Fails gracefully if dependencies are missing

## 🔄 Next Steps

1. **Review and merge** this PR
2. **Test** functionality in a plugin repository release workflow - there is a matching branch in TEC
4. **Configure** plugin-specific changelog URLs in `package.json` files
5. **Monitor** trimming behavior during next release cycle
6. **Adjust** word limits if needed based on WordPress plugin directory feedback

## 📈 Impact

This enhancement will ensure **all TEC plugins maintain WordPress plugin directory compliance** while providing **consistent, professional changelog presentation** across the entire ecosystem. The intelligent trimming preserves user-relevant information while automatically managing file size constraints.

## Test Runs

I have created a [matching branch in TEC](https://github.com/the-events-calendar/the-events-calendar/pull/5276) and have run the action on it. The [action can be seen here](https://github.com/the-events-calendar/the-events-calendar/actions/runs/16478616871) and the [created PR here](https://github.com/the-events-calendar/the-events-calendar/pull/5278/files)

---

*This PR was developed and tested thoroughly locally with real TEC changelog data to ensure reliable functionality across all plugin repositories.* 🎉

### CODE REVIEW

**Review by dpanta94 — APPROVED** [review]

File: .github/actions/process-changelog/README.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+{
+  "tec": {
+    "changelog_url": "https://evnt.is/1b5k"
+  }
+}
+
+// Event Tickets
+{
+  "tec": {
+    "changelog_url": "https://evnt.is/1b5t"
```
Comment (rafsuntaskin) [inline .github/actions/process-changelog/README.md:191]:
```suggestion
    "changelog_url": "https://evnt.is/1b5t"
```
Invalid too.

File: .github/actions/process-changelog/README.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+  },
+  "dependencies": {...}
+}
+```
+
+**Configuration Priority:**
+
+1. **`package.json`** - Plugin-specific configuration (recommended)
+2. **Action parameter** - Workflow-level override
+3. **Default value** - `https://evnt.is/1b5k` fallback
```
[the remark this answers is not in the document]
Comment (rafsuntaskin) [reply .github/actions/process-changelog/README.md:176]: Default may point to regular release [notes](https://theeventscalendar.com/category/release-notes/)

[the remark this answers is not in the document]
Comment (Camwyn) [reply .github/actions/process-changelog/README.md]: Yes - we need shorturls that will point to the plugin's release notes. We also need archives of the release notes by plugin - since sending them to the main release notes page seems excessive.

The TEC PR has a URL specified, but it is also incorrect.

[the remark this answers is not in the document]
Comment (Camwyn) [reply .github/actions/process-changelog/README.md]: Added the enhancement tag so we hold until we get the url issue figured out.

[the remark this answers is not in the document]
Comment (Camwyn) [reply .github/actions/process-changelog/README.md]: Changed to he main release notes URL as a default for now.

File: .github/actions/process-changelog/action.yml
Comment (bordoni) [inline .github/actions/process-changelog/action.yml:19]: Why are we removing the evnt.is link. Create a new link if you need it.

**Review by bordoni — CHANGES_REQUESTED** [review]
Avoid pointing to the real location.
